db/httpdb.js

```diff
@@ -439,6 +440,54 @@ module.exports = function (
     )
   }
 
+  DB.prototype.lockAccount = function (account) {
+    var unlockCode = crypto.randomBytes(16).toString('hex');
+    log.trace({ op: 'DB.lockAccount', uid: account && account.uid, unlockCode: unlockCode })
+
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/lock',
+      {
+        lockedAt: Date.now(),
+        unlockCode: unlockCode
+      }
+    )
+  }
+
+  DB.prototype.unlockAccount = function (account) {
+    log.trace({ op: 'DB.unlockAccount', uid: account && account.uid })
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/unlock'
+      )
+      .then(null,
+        function (err) {
+          if (err.statusCode === 404) {
+            // TODO - throw a better error here.
```

Comment:
In theory, the the only reason this would fail with a 404 is if the account does not exist, so this TODO may be valid.

The SQL is:
```sql
   UPDATE accounts SET lockedAt = null WHERE uid = inUid;
   DELETE FROM accountUnlockCodes WHERE uid = inUid;
```

In the [/account/unlock/verify_code route](https://github.com/mozilla/fxa-auth-server/pull/867/files#diff-43da5bcdd83eb1d30aa0b5e9e7473834R658), we first fetch an account based on uid, then check whether the account is locked. If the account is not locked, we silently succeed without calling unlockAccount.